packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1213,8 +1213,9 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
                     ...options,
                 });
             },
-            pushData: async (...args: Parameters<Dataset['pushData']>) => {
-                return this.pushData(...args);
+            addRequests: this.addRequests.bind(this),
+            pushData: async (data: Parameters<Dataset['pushData']>[0], datasetIdOrName?: string) => {
```

Comment:
this is breaking, isn't it? we were accepting all the arguments, but now you accept only the first one.

Reply:
I sure hope it's not. As far as I know, there is only one parameter in `Dataset.pushData`

Reply:
It was a bit of a "while I'm at it" moment. It made sense to me to include this in the restricted context (to prevent a possible case when a user may need to unrestrict themself) and so I propagated it to the original one as well.